### ISSUE TITLE
Blockly should allow developper to add auto-dynamic tools category

### ISSUE THREAD

**Issue by fredlefred** [google/blockly#259]
There's two dynamic tools category : variable and procedure, width little work it will be easy to allow other dynamic tools category:

see joined diff.
[flyout.js.diff.txt](https://github.com/google/blockly/files/138218/flyout.js.diff.txt)

With this diff, any developper can add dynamic tools category by using : 

`Blockly.registerFlyoutCategory('MY_CUSTOM_TOOLS',flyoutMyCustomToolsCategory) ;
`

The function flyoutMyCustomToolsCategory must return an array of already register 'block' like :

```
var flyoutMyCustomToolsCategory = function(workspace) {
  var xmlList = [];
  if (Blockly.Blocks['customBlock1']) {
    // <block type="procedures_defnoreturn" gap="16"></block>
    var block = goog.dom.createDom('block');
    block.setAttribute('type', 'customBlock1');
    block.setAttribute('gap', 8);
    xmlList.push(block);
  }
  if (Blockly.Blocks['customBlock2']) {
    // <block type="procedures_defnoreturn" gap="16"></block>
    var block = goog.dom.createDom('block');
    block.setAttribute('type', 'customBlock2');
    block.setAttribute('gap', 32);
    xmlList.push(block);
  }
 // and so on, it can be dynamic, it is called every time this tools category is open.
 return xmlList ;
}
```

Then in xml tools developper should add :
`<category name="Custom Blocks" id="catCustomBlocks" colour="290" custom="MY_CUSTOM_TOOLS"></category>
`

**Comment by violalbi91** [comment]
Hi,

I'm interested in this topic, where I have to put the line
`Blockly.registerFlyoutCategory('MY_CUSTOM_TOOLS',flyoutMyCustomToolsCategory) ; ` ?

Do I have to eliminate the part in .show()

`if (xmlList == Blockly.Variables.NAME_TYPE) {
    // Special category for variables.
    Blockly.Variables.flyoutCategory(blocks, gaps, margin,
        /** @type {!Blockly.Workspace} */ (this.workspace_));
  } else if (xmlList == Blockly.Procedures.NAME_TYPE) {
    // Special category for procedures.
    Blockly.Procedures.flyoutCategory(blocks, gaps, margin,
        /** @type {!Blockly.Workspace} */ (this.workspace_));`

with the last part of the code inside the file flout.s.diff.txt?

**Comment by rachel-fenichel** [comment]
Fixed by #863
